### ISSUE TITLE
Support exposing eq and hash to Dart of opaque Rust object

### ISSUE THREAD

**Issue by boxdot** [phnx-im/infra#247]
Currently, opaque Rust objects are compared by identity in Rust. When such objects are used as cubit state, the UI might redraw more often than needed: every time a cubit emits a state which is the same in terms of equality, the UI will still rebuild because the emitted object is not the same. To avoid that, we need to be able to override `operator ==` in Dart by Rust's `eq` impl. This is currently not possible.

See the following discussion: https://github.com/fzyzcjy/flutter_rust_bridge/issues/2238

**Comment by dbsxdbsx** [comment]
Is there any progress on this issue?  I've not paid attention to the `eq` behavior consistency across Rust and Dart for a long time, I don't know if this feature has been implemented or not.

**Comment by boxdot** [comment]
> Is there any progress on this issue? I've not paid attention to the `eq` behavior consistency across Rust and Dart for a long time, I don't know if this feature has been implemented or not.

No, PartialEq impl is not transported from Rust to Dart. In Dart comparison is done via Object-identity.
